### PR TITLE
Initial implementation of the extension

### PR DESCRIPTION
**Pull request by mmorhun**
This extension contributes to `HostedPluginManager` uri postprocessor which converts internal (inside container) hosted instance uri to external (Che host).

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,31 @@
+{
+  "name": "@theia/che-theia-hosted-plugin-manager-extension",
```
Comment (benoitf) [inline package.json]: hello, why is it @theia if it's a che extension ?

File: tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+            "dom"
+        ],
+        "sourceMap": true,
+        "rootDir": "src",
+        "outDir": "lib"
+    },
+    "include": [
+        "src"
+    ]
+}
```
Comment (benoitf) [inline tsconfig.json]: there is a missing EOF

File: src/node/che-workspace-hosted-plugin-uri-post-processor.ts [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        const token = process.env.CHE_MACHINE_TOKEN;
+        if (token) {
+            restAPIConfig.headers = {};
+            restAPIConfig.headers['Authorization'] = token;
+        }
+        this.restApiClient = WorkspaceClient.getRestApi(restAPIConfig);
+    }
+
+    async processUri(uri: URI): Promise<URI> {
+        const hostedPluginTheiaInnstanceServer = await this.getHostedPluginTheiaInnstanceServer();
```
Comment (benoitf) [inline src/node/che-workspace-hosted-plugin-uri-post-processor.ts]: somehow there are too many n in 'instance' getHostedPluginTheiaInnstanceServer

File: tsconfig.json (hunk 2 above)
Comment (mmorhun) [reply tsconfig.json]: will fix

File: src/node/che-workspace-hosted-plugin-uri-post-processor.ts (hunk 3 above)
Comment (mmorhun) [reply src/node/che-workspace-hosted-plugin-uri-post-processor.ts]: Thanks

File: package.json (hunk 1 above)
Comment (mmorhun) [reply package.json]: will change for `eclipse-che`

File: package.json [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    "node"
+  ],
+  "dependencies": {
+    "@theia/core": "latest",
+    "@theia/plugin-ext": "latest",
+    "@eclipse-che/workspace-client": "latest"
+  },
+  "devDependencies": {
+    "rimraf": "^2.6.2",
+    "typescript": "latest"
```
Comment (AndrienkoAleksandr) [inline package.json]: Set up please 2.7.2 version. For this one version CQ passed.

Comment (mmorhun) [reply package.json]: ok

**Pull request merged**